client/core/core.go

```diff
@@ -148,13 +152,18 @@ func (dc *dexConnection) refreshMarkets() map[string]*Market {
 		}
 		mid := market.marketName()
 		dc.tradeMtx.RLock()
+		mktTrades := make([]*trackedTrade, 0, len(dc.trades))
 		for _, trade := range dc.trades {
 			if trade.mktID == mid {
-				corder, _ := trade.coreOrder()
-				market.Orders = append(market.Orders, corder)
+				mktTrades = append(mktTrades, trade)
 			}
 		}
 		dc.tradeMtx.RUnlock()
+
+		for _, trade := range mktTrades {
+			corder, _ := trade.coreOrder()
+			market.Orders = append(market.Orders, corder)
+		}
```

Comment:
There's a good deal of lock contention for `dc.tradeMtx`  I've reduced that in a few spots where too much happens while the lock is held.